LEAN-3555: LW-FY24-Q4-S6: Inability to Edit New Paragraph in General …

nghazzawi-atypon opened the pull request:
…Note After Insertion with Enter Key

asouqi commented:
@nghazzawi-atypon I can't reproduce LEAN-3555, do you have an XML article example with that issue? and in this PR you are trying to fix the issue of Enter in the general footnote paragraph?

nghazzawi-atypon commented:
> @nghazzawi-atypon I can't reproduce LEAN-3555, do you have an XML article example with that issue? and in this PR you are trying to fix the issue of Enter in the general footnote paragraph?

The issue no longer occurs after implementing LEAN-3519. However, to activate this feature or resolve the current ticket, some tasks remain: styling adjustments for selected and un-selected nodes, code cleanup, and updating how nodes are being selected and deleted and that's what was done here so far.

asouqi commented:
> @asouqi
> 
> > @nghazzawi-atypon I can't reproduce LEAN-3555, do you have an XML article example with that issue? and in this PR you are trying to fix the issue of Enter in the general footnote paragraph?
> 
> The issue no longer occurs after implementing LEAN-3519. However, to activate this feature or resolve the current ticket, some tasks remain: styling adjustments for selected and un-selected nodes, code cleanup, and updating how nodes are being selected and deleted and that's what was done here so far.

please check the case of insert table footnote, fn element border will not be clear:
![Screenshot from 2024-07-21 16-10-41](https://github.com/user-attachments/assets/392ffac5-dccc-49e3-8c02-092a6284da27)

nghazzawi-atypon commented:
> > @asouqi
> > > @nghazzawi-atypon I can't reproduce LEAN-3555, do you have an XML article example with that issue? and in this PR you are trying to fix the issue of Enter in the general footnote paragraph?
> > 
> > 
> > The issue no longer occurs after implementing LEAN-3519. However, to activate this feature or resolve the current ticket, some tasks remain: styling adjustments for selected and un-selected nodes, code cleanup, and updating how nodes are being selected and deleted and that's what was done here so far.
> 
> please check the case of insert table footnote, fn element border will not be clear: ![Screenshot from 2024-07-21 16-10-41](https://private-user-images.githubusercontent.com/45534101/350770360-392ffac5-dccc-49e3-8c02-092a6284da27.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE1ODczMjEsIm5iZiI6MTcyMTU4NzAyMSwicGF0aCI6Ii80NTUzNDEwMS8zNTA3NzAzNjAtMzkyZmZhYzUtZGNjYy00OWUzLThjMDItMDkyYTYyODRkYTI3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzIxVDE4MzcwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFmZGRkNDE2Yjk1YzkzY2ZmYzdhMzA1M2QwZDZhMzk3NjI1MDZkNTY1NzhkN2VlYjNhNzFkMmNiYWYwMjkzMzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.TrwseOZhVWhq1hf49ctmOy0Zdj5MdIXfqXh9I72YMhs)

@asouqi 
This issue is already in the master branch, but I've fixed it anyway.
![image](https://github.com/user-attachments/assets/bf17e8f4-0c94-4749-a7ac-972a968aad28)
